apps/comps/components/navbar/Claim.tsx

```diff
@@ -1,18 +1,60 @@
+import * as dnum from "dnum";
+import { useEffect } from "react";
+
 import { Button } from "@recallnet/ui2/components/button";
+import { toast } from "@recallnet/ui2/components/toast";
 
 import { Recall } from "@/components/Recall";
-import { formatAmount } from "@/utils/format";
+import { useClaim } from "@/hooks/useClaim";
 
 export const Claim = () => {
-  const value = 3000;
+  const { totalClaimable, claim, isPending, isConfirming, isConfirmed, error } =
+    useClaim();
+
+  const display = dnum.format([totalClaimable, 18], { compact: true });
+
+  const isProcessing = isPending || isConfirming;
+  const disabled = totalClaimable === 0n || isProcessing;
+
+  // Show success toast when claim is confirmed
+  useEffect(() => {
+    if (isConfirmed) {
+      toast.success("Successfully claimed rewards!");
+    }
+  }, [isConfirmed]);
+
+  // Show error toast when there's an error
+  useEffect(() => {
+    if (error) {
+      toast.error("Failed to claim rewards");
+      console.error("Claim error:", error);
+    }
+  }, [error]);
+
+  const handleClaim = async () => {
+    try {
+      await claim(undefined);
```

Comment:
what is this `undefined` here?

Reply:
Imagine that you could omit `undefined`, right?

Reply:
changed in https://github.com/recallnet/js-recall/pull/1485/commits/caedc851c879e4ff97568387a1724889373a4f28